firestore/query.go

```diff
@@ -549,22 +536,26 @@ func (q *Query) fieldValuesToCursorValues(fieldValues []interface{}) ([]*pb.Valu
 	for i, ord := range q.orders {
 		fval := fieldValues[i]
 		if ord.isDocumentID() {
-			// TODO(jba): support DocumentRefs as well as strings.
 			// TODO(jba): error if document ref does not belong to the right collection.
-			docID, ok := fval.(string)
-			if !ok {
+
+			switch docID := fval.(type) {
+			case string:
+				vals[i] = &pb.Value{ValueType: &pb.Value_ReferenceValue{q.path + "/" + docID}}
+				continue
+			case *DocumentRef:
+				// DocumentRef can be transformed in usual way.
+			default:
 				return nil, fmt.Errorf("firestore: expected doc ID for DocumentID field, got %T", fval)
```

Comment:
I could change this err, but it doesn't seem far off since we accept docRefs now in addition. Happy to alter though if anyone has opinions.